test/e2e/framework/framework.go

```diff
@@ -39,6 +39,19 @@ const (
 	serviceAccountTokenSecretWaitTimeout = 1 * time.Minute
 )
 
+type FrameworkSet []*Framework
+
+func NewFrameworkSet(generateName string) FrameworkSet {
+	fs := make(FrameworkSet, 0, len(TestContext.RestConfigs))
+
+	uniqueName := names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", generateName))
+	for _, restConfig := range TestContext.RestConfigs {
+		fs = append(fs, NewFramework(uniqueName, WithRestConfig(restConfig)))
```

Comment:
restConfig is a required arg, let's just put it as an arg to the function constructor so its type safe and nobody can forget to give it to the constructor

Reply:
the issue with this imo is that so far it was defaulted, so suddenly every existing test has to pass a rest config to the framework constructor, which not only requires changes to every single test, but sort of brakes the separation of concerns?

Reply:
nevermind, works well with the cluster idea from the other comment